libsolidity/parsing/Parser.cpp

```diff
@@ -1148,7 +1148,7 @@ ASTPointer<InlineAssembly> Parser::parseInlineAssembly(ASTPointer<ASTString> con
 	SourceLocation location{position(), -1, source()};
 
 	expectToken(Token::Assembly);
-	yul::Dialect const& dialect = yul::EVMDialect::looseAssemblyForEVM(m_evmVersion);
+	yul::Dialect const& dialect = yul::EVMDialect::strictAssemblyForEVM(m_evmVersion);
```

Comment:
This line change causes some tests to fail, for example: CREATE2 is being tested even though it's not present in EVM version homestead.